tests/thread_race/main.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     test
+ * @{
+ *
+ * @file
+ * @brief       Test for race conditions in context switching
+ *
+ * May have false negatives.
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include <avr/cpufunc.h>
+#include <avr/interrupt.h>
+#include <stdio.h>
+
+#include "irq.h"
+#include "sched.h"
+#include "thread.h"
+
+/* board.h for AVR_CONTEXT_SWAP_TRIGGER */
+#include "board.h"
```

Comment:
Here and below, I don't like the non-generic nature on this tests, and why AVR should be treated in such special way... I see more convenient to have a generic test which works for all platforms without `#ifdef`s.

If this is due to #8896, please make this PR dependent on #8904, on which `AVR_CONTEXT_SWAP_TRIGGER` doesn't exist anymore.